dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/persistence/file/advanced/ConsentAwareFileOrchestratorTest.kt

```diff
@@ -0,0 +1,532 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.file.advanced
+
+import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+import com.datadog.android.core.internal.privacy.ConsentProvider
+import com.datadog.android.privacy.TrackingConsent
+import com.datadog.android.utils.forge.Configurator
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.reset
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.io.File
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class ConsentAwareFileOrchestratorTest {
+
+    lateinit var testedOrchestrator: ConsentAwareFileOrchestrator
+
+    @Mock
+    lateinit var mockConsentProvider: ConsentProvider
+
+    @Mock
+    lateinit var mockPendingOrchestrator: FileOrchestrator
+
+    @Mock
+    lateinit var mockGrantedOrchestrator: FileOrchestrator
+
+    @Mock
+    lateinit var mockDataMigrator: DataMigrator
+
+    @BeforeEach
+    fun `set up`() {
+        instantiateTestedOrchestrator(TrackingConsent.PENDING)
+        reset(mockDataMigrator)
+    }
+
+    // region init
+
+    @Test
+    fun `𝕄 migrate data 𝕎 init() {GRANTED}`() {
+        // When
+        instantiateTestedOrchestrator(TrackingConsent.GRANTED)
+
+        // Then
+        verify(mockDataMigrator).migrateData(
+            null,
+            mockPendingOrchestrator,
+            TrackingConsent.GRANTED,
+            mockGrantedOrchestrator
+        )
+    }
+
+    @Test
+    fun `𝕄 migrate data 𝕎 init() {PENDING}`() {
+        // When
+        instantiateTestedOrchestrator(TrackingConsent.PENDING)
+
+        // Then
+        verify(mockDataMigrator).migrateData(
+            null,
+            mockPendingOrchestrator,
+            TrackingConsent.PENDING,
+            mockPendingOrchestrator
+        )
+    }
+
+    @Test
+    fun `𝕄 migrate data 𝕎 init() {NOT_GRANTED}`() {
```

Comment:
The class can't be instantiated with a `null` value (the initial consent is taken from `ConsentProvider.getConsent()` which returns a non null value).

Reply:
but there is a branch in the `ConsentAwareFileOrchestrator` which has `null`:

https://github.com/DataDog/dd-sdk-android/pull/542/files#diff-0f67e9256bb45f7ebef96fc812346070fb3a8dc12003026de6cd87e80901b34fR79

Reply:
That's because this method is called with null on `init`: 
```
handleConsentChange(null, consentProvider.getConsent())
```
Those are handled by tests in the init section